In [2]:
import pandas as pd
df = pd.read_csv('/content/DBS_SingDollar.csv')
df

Unnamed: 0        Date    DBS      SGD
0             1   6/20/2017  20.45  1.38925
1             2   6/19/2017  20.60  1.38707
2             3   6/16/2017  20.45  1.38324
3             4   6/15/2017  20.41  1.38378
4             5   6/14/2017  20.59  1.37595
..          ...         ...    ...      ...
117         118  12/30/2016  17.09  1.44655
118         119  12/29/2016  17.13  1.44814
119         120  12/28/2016  17.22  1.45018
120         121  12/27/2016  17.19  1.44954
121         122  12/23/2016  17.14  1.44766

[122 rows x 4 columns]

In [3]:
# Set 'SGD' as Independent Variable (Exchange Rate) and 'DBS' as Dependent Variable (Stock Price)
X = df.loc[:,['SGD']] # Selects all rows and the column 'SGD', store result as DataFrame
Y = df.loc[:,['DBS']] # Selects all rows and the column 'DBS', store result as DataFrame

In [5]:
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

# Using linear regression for prediction
model = LinearRegression()
model.fit(X,Y)
prediction = model.predict(X)
rmse = mean_squared_error(Y,prediction)**0.5
print(rmse)

0.6283698867042165


In [6]:
import joblib # Used to save the trained model into a pickle file
joblib.dump(model,'DBS_SGD_Prediction.pkl')

['DBS_SGD_Prediction.pkl']

In [8]:
model.coef_

array([[-50.60094302]])

- This is the slope (weight) learned by the model

- Since you used one feature (SGD), there is one coefficient

**Interpretation**:

> For every 1 unit increase in SGD, the predicted DBS value decreases by ~50.6 units

In [9]:
model.intercept_

array([90.22858515])

- This is the bias term

- Value of DBS when SGD = 0 (theoretical baseline)

The full prediction equation

The model represents:

`DBS = (−50.6009) × SGD + 90.2286`
